tests/collecting/CollectorsTest.py

```diff
@@ -202,20 +200,18 @@ def setUp(self):
         self.collectors_test_dir = os.path.join(current_dir,
                                                 "collectors_test_dir")
 
-        self.log_printer = LogPrinter(ConsolePrinter())
+        self.log_printer = ListLogPrinter()
 
     def test_bear_empty(self):
         self.assertRaises(TypeError, collect_bears)
 
     def test_bear_invalid(self):
-        with retrieve_stdout() as sio:
-            self.assertEqual(collect_bears(["invalid_paths"],
-                                           ["invalid_name"],
-                                           ["invalid kind"],
-                                           self.log_printer), ([],))
-            self.assertRegex(sio.getvalue(),
-                             ".*\\[WARNING\\].*No bears were found matching "
-                             "'invalid_name'.\n")
+        self.assertEqual(collect_bears(["invalid_paths"],
+                                       ["invalid_name"],
+                                       ["invalid kind"],
+                                       self.log_printer), ([],))
+        self.assertEqual(''.join(log.message for log in self.log_printer.logs),
+                         "No bears were found matching 'invalid_name'.")
```

Comment:
how about `self.assertEqual([log.message for log in self.log_printer], ['No ...'])`?